source/request_response.c

```diff
@@ -317,6 +317,23 @@ struct aws_http_stream *aws_http_stream_new_client_request(const struct aws_http
     return stream;
 }
 
+int aws_http_stream_configure_server_request_handler(
+    struct aws_http_stream *stream,
+    const struct aws_http_request_handler_options *options) {
+    if (!options || options->self_size == 0 || !options->server_connection) {
```

Comment:
You should also check that this function is only called once ever, `aws_http_connection_configure_server()` does a similar check